dowhy/causal_estimator.py

```diff
@@ -231,67 +212,278 @@ def get_confidence_intervals(self, estimate, num_ci_simulations = None, sample_s
+            causal estimator.
+        '''
+        raise NotImplementedError
 
-        Uses resampling to create a non-parametric significance test.
-        A general procedure. Individual estimators can override this method.
+    def estimate_confidence_intervals(self, confidence_level=None, method=None, 
+             **kwargs):
+        '''
+            Find the confidence intervals corresponding to any estimator
+            By default, this is done with the help of bootstrapped confidence intervals
+            but can be overridden if the specific estimator implements other methods of estimating confidence intervals. 
 
-        :param self: object instance of class Estimator
-        :param estimate: obtained estimate
-        :param num_simulations: (optional) number of simulations to run
-        :returns:
+            If the method provided is not bootstrap, this function calls the implementation of the specific estimator.
 
+            :param method: Method for estimating confidence intervals. 
+            :param confidence_level: The confidence level of the confidence intervals of the estimate.
+            :param kwargs: Other optional args to be passed to the CI method.
+            :returns: The obtained confidence interval. 
+        '''
+        if method is None:
+            if self._confidence_intervals:
+                method = self._confidence_intervals # this is either True or methodname
+            else:
+                method = "default"
+        confidence_intervals = None
+        if confidence_level is None:
+            confidence_level = self.confidence_level
+        if method == "default" or method is True: # user has not provided any method
+            try:
+                confidence_intervals = self._estimate_confidence_intervals(
+                        confidence_level, method=method, **kwargs)
+            except NotImplementedError:
+                confidence_intervals = self._estimate_confidence_intervals_with_bootstrap(
+                         confidence_level, **kwargs)
+        else:
+            if method == "bootstrap":
+                confidence_intervals = self._estimate_confidence_intervals_with_bootstrap(
+                         confidence_level, **kwargs)
+            else:
+                confidence_intervals = self._estimate_confidence_intervals(
+                        confidence_level, method=method, **kwargs)
+        return confidence_intervals
+        
+    def _estimate_std_error_with_bootstrap(self, num_ci_simulations=None,
+            sample_size_fraction=None):
+        """ Compute standard error using the bootstrap method.
+
+        :param num_simulations: Number of bootstrapped samples.
+        :param sample_size_fraction: Fraction of data to be resampled. 
+        :returns: Standard error of the obtained estimate.
+        """
+        # Use existing params, if new user defined params are not present
+        if num_ci_simulations is None:
```

Comment:
I believe this should be ```num_null_simulations``` corresponding to Line 184

Reply:
yes good point. I thought about this decision initially too, and decided to keep a single parameter for both ci and std error since they both can be computed from the same bootstrap. But I see your point--this seems odd and confusing.

How about renaming num_ci_simulations to simply num_simulations? that way we can reuse both.

Reply:
i worry that we have support more estimates in the future, e.g., error variance or R2 and it would hurt to have a separate simulation parameter for each.

Reply:
How about this, we have a global attribute like ```num_simulations``` to hold the default number of simulations or the number of simulations to be set for all functions. We can use local variables corresponding to each scenario.

Reply:
yes, the current implementation supports that. Marking as resolved. thanks for the comments!